py_async_grpc_prometheus/grpc_utils.py

```diff
@@ -1,21 +1,35 @@
 import grpc
+from typing import Iterable, AsyncIterable
 UNARY = "UNARY"
 SERVER_STREAMING = "SERVER_STREAMING"
 CLIENT_STREAMING = "CLIENT_STREAMING"
 BIDI_STREAMING = "BIDI_STREAMING"
 UNKNOWN = "UNKNOWN"
 
 
-async def wrap_iterator_inc_counter(iterator, counter, grpc_type, grpc_service_name, grpc_method_name):
+async def wrap_iterator_inc_counter(iterator, counter, grpc_type, grpc_service_name, grpc_method_name, observer = None, legacy_observer = None, timer = None, start = None):
   """Wraps an iterator and collect metrics."""
-  
-  async for item in iterator:
-    counter.labels(
-      grpc_type=grpc_type,
-      grpc_service=grpc_service_name,
-      grpc_method=grpc_method_name).inc()
-    yield item
 
+  if isinstance(iterator, Iterable):
+    for item in iterator:
+      counter.labels(
+        grpc_type=grpc_type,
+        grpc_service=grpc_service_name,
+        grpc_method=grpc_method_name).inc()
+      yield item
+  elif isinstance(iterator, AsyncIterable):
+    async for item in iterator:
+      counter.labels(
+        grpc_type=grpc_type,
+        grpc_service=grpc_service_name,
+        grpc_method=grpc_method_name).inc()
+      yield item
+  if timer:
```

Comment:
Why is the timer necessary?

Reply:
The timer is to record the total time the call took for the server streaming to finish. I'm trying for parity with go interceptors: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/interceptors/client.go

As I read this, the timer finishes after the EOF is sent, which is the same as when the iterator would stop iterating in python.